File: Difficulty/Easy/[66] Plus One/solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if not digits:
            return [1]
        carry = 0
        digits[len(digits) - 1] += 1
        for d in range(len(digits) - 1, -1, -1):
            digits[d] += carry
            if digits[d] != 10:
                carry = 0
                break
            else:
                carry = 1
                digits[d] = 0
        if carry == 1:
            digits.append(1)
            digits[0], digits[len(digits) - 1] = digits[len(digits) - 1], digits[0]
        return digits

File: Difficulty/Easy/[66] Plus One/solution.cpp
/*
 * @lc app=leetcode id=66 lang=cpp
 *
 * [66] Plus One
 */
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        if (digits.empty()) return vector<int>(1,1);
        int carry = 0;
        digits[digits.size() - 1] += 1;
        for (auto it = digits.rbegin(); it != digits.rend(); it++){
            *it += carry;
            if (*it != 10){
                carry = 0;
                break;
            }
            else{
                carry = 1;
                *it = 0;
            }
        }
        if (carry == 1){
            digits.push_back(1);
            swap(digits[0], digits[digits.size() - 1]);
            return digits;
        }
        else
            return digits;
    }
};

